src/roiextractors/extractors/memmapextractors/memmapextractors.py

```diff
@@ -27,17 +27,19 @@ def __init__(
         self._video = video
         super().__init__()
 
-    def get_frames(self, frame_idxs=None):
+    def get_frames(self, frame_idxs=None, channel: int = 0):
         if frame_idxs is None:
             frame_idxs = [frame for frame in range(self.get_num_frames())]
 
         frames = self._video.take(indices=frame_idxs, axis=0)
+        if channel is not None:
+            frames = frames[:, :, :, channel].squeeze()
 
         return frames
 
-    def get_video(self, start_frame: int = None, end_frame: int = None) -> np.array:
+    def get_video(self, start_frame: int = None, end_frame: int = None, channel: int = 0) -> np.array:
```

Comment:
Since this is getting handed off, essentially, to `get_frames`, mimic requested annotation changes here as well

Reply:
Also, recommend annotation typing `np.array` as `np.ndarray` - interesting read: https://stackoverflow.com/questions/35673895/type-hinting-annotation-pep-484-for-numpy-ndarray

Reply:
Good. I took the chance to add a bunch of typing information together with this. Check the latest commit.